File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.go
func countCompleteSubarrays(nums []int) int {
    s := make(map[int]int) 
    for _, num := range nums {
        s[num]++
    }
    
    completeSize := len(s) 
    left := 0
    elems := 0
    ans := 0
    freq := make(map[int]int)
    for right, num := range nums {
        freq[num]++
        if freq[num] == 1 {
            elems++
        }
        for left <= right && elems == completeSize {
            freq[nums[left]]--;
            if freq[nums[left]] == 0 {
                elems--
            }
            left++
        }

        ans += left
    }
    return ans
}

File: 2856-count-complete-subarrays-in-an-array/2856-count-complete-subarrays-in-an-array.cpp
class Solution {
   public:
    int countCompleteSubarrays(vector<int>& nums) {
        unordered_set<int> s;
        for (const auto& num : nums) s.insert(num);

        int n = nums.size();
        int left = 0;
        int completeCount = s.size();
        int elems = 0;
        int ans = 0;
        unordered_map<int, int> freq;
        for (int right = 0; right < n; right++) {
            freq[nums[right]]++;
            if (freq[nums[right]] == 1) elems++;

            while (left <= right && elems == completeCount) {
                freq[nums[left]]--;
                if (freq[nums[left]] == 0) elems--;
                left++;
            }
            ans += left;
        }
        return ans;
    }
};